docs/index.rst

```diff
@@ -125,5 +125,5 @@ Fifth-level heading
 This is the deepest heading level we support.
 
 
-.. _developer guide: https://github.com/crate/crate-docs-theme/blob/master/DEVELOP.rst#make-changes
+.. _developer guide: https://github.com/crate/crate-docs-theme/blob/master/DEVELOP.rst#user-content-make-changes
```

Comment:
Attaching to this, I just created https://github.com/sphinx-doc/sphinx/issues/9016.